File: apply_template.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import re
import json

# 讀取data.js
with open('js/data.js', 'r', encoding='utf-8') as f:
    content = f.read()

# 26位Vtuber ID（除了shiucoda）
vtuber_ids = [
    'rumi',  # 已修改
    'paroniie', 'lubee', 'himegimichika', 'arrynia', 'kiriko',
    'vaswawa', 'sinniearis', 'ririna', 'ekorru', 'wakasaito', 'whalefall', 'kai',
    'mukuru', 'nyrfier', 'fuka22', 'chita', 'chamamatti', 'shiroleon',
    'BarkBarkPomi', 'cocor0', 'UchiFifi', 'nokori', 'KeKeMii', 'Pele', 'Yawnii'
]

new_videos_struct = '''    videos: {
      covers: [],
      originals: [],
      shorts: [],
      general: [],
      premiere: [],
      vlog: [],
      ads: [],
      commerce: [],
      unclassified: []
    },
    memberVideos: [],'''

processed = 0
skipped = 0

print("開始修改data.js...")

for vid in vtuber_ids:
    # 檢查是否已經有新格式
    if f'id: "{vid}"' in content:
        # 提取該Vtuber的段落（使用更精確的正則表達式）
        # 尋找該Vtuber到下一個 `},` 或 `},\n  {`
        pattern = rf'(id:\s*"{vid}"[^}}]*?)(videos:\s*\{{[\s\S]*?\}}|videos:\s*\[[^\]]*\])(,)'

        match = re.search(pattern, content)
        if match:
            # 檢查是否已經是新格式
            if 'covers:' in content[content.find(f'id: "{vid}"'):content.find(f'id: "{vid}"') + 1000]:
                print(f"⏭️  {vid} 已是新格式，跳過")
                skipped += 1
                continue

            # 替換
            before = content[:match.start(3)]
            after = content[match.end(3):]
            content = before + new_videos_struct + ',\n' + after

            print(f"✅ {vid} 已修改")
            processed += 1

# 寫回檔案
with open('js/data.js', 'w', encoding='utf-8') as f:
    f.write(content)

print(f"\n✨ 完成！")
print(f"✅ 已修改: {processed}")
print(f"⏭️  已跳過: {skipped}")

File: scripts/apply-template.js
#!/usr/bin/env node

/**
 * 自動為26位Vtuber應用詩雨蔻達的完整範本
 * 功能：
 * 1. 為每位Vtuber添加完整的videos分類結構
 * 2. 為每位Vtuber添加memberVideos空陣列
 * 3. 保留詩雨蔻達不變
 */

const fs = require('fs');
const path = require('path');

const dataPath = path.join(__dirname, '../js/data.js');

// 讀取data.js
let content = fs.readFileSync(dataPath, 'utf8');

// 定義的26位Vtuber ID（除了詩雨蔻達）
const vtuberIds = [
  'rumi', 'paroniie', 'lubee', 'himegimichika', 'arrynia', 'kiriko',
  'vaswawa', 'sinniearis', 'ririna', 'ekorru', 'wakasaito', 'whalefall', 'kai',
  'mukuru', 'nyrfier', 'shiroleon', 'fuka22', 'BarkBarkPomi', 'cocor0', 'UchiFifi',
  'chita', 'nokori', 'KeKeMii', 'chamamatti', 'Pele', 'Yawnii'
];

// 為每位Vtuber應用範本
let processed = 0;
vtuberIds.forEach(vId => {
  // 使用正則表達式找到該Vtuber的完整物件
  const vtuberRegex = new RegExp(
    `(\\{[\\s\\S]*?id:\\s*["']${vId}["'][\\s\\S]*?)(?=(,\\s*\\{\\s*id:|\\];))`,
    'g'
  );

  const match = vtuberRegex.exec(content);
  if (!match) {
    console.log(`❌ 找不到Vtuber: ${vId}`);
    return;
  }

  let vtuberObj = match[1];

  // 檢查是否已經有完整的分類結構
  if (!vtuberObj.includes('covers:')) {
    // 添加完整的分類結構
    const template = `
    covers: [],          // Cover/歌ってみた
    originals: [],       // Original/原創
    shorts: [],          // Shorts（≤60秒）
    general: [],         // 一般影片
    vlog: [],            // Vlog（日常生活內容）
    ads: [],             // 廣告
    premiere: [],        // 初配信
    unclassified: [],    // 未分類（待手動分類）
    memberVideos: [],    // 會員直播（待手動新增）`;

    // 移除原有的videos陣列或空陣列
    const oldVideosRegex = /videos:\s*(\[[\s\S]*?\]|null|undefined)/;
    vtuberObj = vtuberObj.replace(oldVideosRegex, '');

    // 移除原有的memberVideos（如果有的話）
    const oldMemberVideosRegex = /,?\s*memberVideos:\s*\[[\s\S]*?\]/g;
    vtuberObj = vtuberObj.replace(oldMemberVideosRegex, '');

    // 在youtubeChannelId後插入新的分類結構
    const insertPoint = vtuberObj.lastIndexOf(',');
    if (insertPoint > 0) {
      vtuberObj = vtuberObj.slice(0, insertPoint) + ',' + template + vtuberObj.slice(insertPoint + 1);
    }

    // 替換整個Vtuber物件
    const beforeMatch = content.substring(0, match.index);
    const afterMatch = content.substring(match.index + match[0].length);
    content = beforeMatch + vtuberObj + afterMatch;

    console.log(`✅ 應用範本成功: ${vId}`);
    processed++;
  } else {
    console.log(`⏭️  ${vId} 已經有完整分類結構，跳過`);
  }
});

// 保存修改
fs.writeFileSync(dataPath, content, 'utf8');
console.log(`\n✅ 完成！已為 ${processed} 位Vtuber應用範本`);
